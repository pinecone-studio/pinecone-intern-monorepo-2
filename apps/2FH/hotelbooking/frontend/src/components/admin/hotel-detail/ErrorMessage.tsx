'use client';

interface ErrorMessageProps {
  message: string;
}

export const ErrorMessage = ({ message }: ErrorMessageProps) => {
  return (
    <div className="min-h-screen bg-gray-50 p-6">
      <div className="max-w-6xl mx-auto">
        <div className="bg-red-50 border border-red-200 text-red-700 px-4 py-3 rounded">
          Error loading hotel: {message}
        </div>
      </div>
    </div>
  );
}; 