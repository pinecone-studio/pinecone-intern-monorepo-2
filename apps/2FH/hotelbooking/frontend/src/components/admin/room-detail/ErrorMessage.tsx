import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { AlertCircle } from 'lucide-react';

interface ErrorMessageProps {
  message: string;
}

export const ErrorMessage = ({ message }: ErrorMessageProps) => {
  return (
    <div className="min-h-screen bg-gray-50 p-6">
      <div className="max-w-6xl mx-auto">
        <Card>
          <CardHeader>
            <CardTitle className="flex items-center gap-2 text-red-600">
              <AlertCircle className="h-5 w-5" />
              Error
            </CardTitle>
          </CardHeader>
          <CardContent>
            <p className="text-gray-700">{message}</p>
          </CardContent>
        </Card>
      </div>
    </div>
  );
};


