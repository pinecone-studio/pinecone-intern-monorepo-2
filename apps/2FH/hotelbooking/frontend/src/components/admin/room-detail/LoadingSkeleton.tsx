import { Card, CardContent, CardHeader } from '@/components/ui/card';

export const LoadingSkeleton = () => {
  return (
    <div className="min-h-screen bg-gray-50 p-6">
      <div className="max-w-6xl mx-auto">
        {/* Header Skeleton */}
        <div className="flex items-center gap-4 mb-8">
          <div className="h-10 w-20 bg-gray-200 rounded animate-pulse"></div>
          <div className="space-y-2">
            <div className="h-8 w-64 bg-gray-200 rounded animate-pulse"></div>
            <div className="h-4 w-32 bg-gray-200 rounded animate-pulse"></div>
          </div>
        </div>

        <div className="grid grid-cols-1 lg:grid-cols-2 gap-6">
          {/* Left Side Skeleton */}
          <div className="space-y-6">
            <Card>
              <CardHeader>
                <div className="h-6 w-32 bg-gray-200 rounded animate-pulse"></div>
              </CardHeader>
              <CardContent className="space-y-4">
                <div className="h-4 w-full bg-gray-200 rounded animate-pulse"></div>
                <div className="h-4 w-3/4 bg-gray-200 rounded animate-pulse"></div>
                <div className="h-4 w-1/2 bg-gray-200 rounded animate-pulse"></div>
              </CardContent>
            </Card>

            <Card>
              <CardHeader>
                <div className="h-6 w-40 bg-gray-200 rounded animate-pulse"></div>
              </CardHeader>
              <CardContent className="space-y-4">
                <div className="h-4 w-full bg-gray-200 rounded animate-pulse"></div>
                <div className="h-4 w-2/3 bg-gray-200 rounded animate-pulse"></div>
              </CardContent>
            </Card>

            <Card>
              <CardHeader>
                <div className="h-6 w-36 bg-gray-200 rounded animate-pulse"></div>
              </CardHeader>
              <CardContent>
                <div className="grid grid-cols-2 gap-4">
                  <div className="h-20 bg-gray-200 rounded animate-pulse"></div>
                  <div className="h-20 bg-gray-200 rounded animate-pulse"></div>
                </div>
              </CardContent>
            </Card>
          </div>

          {/* Right Side Skeleton */}
          <div className="space-y-6">
            <Card>
              <CardHeader>
                <div className="h-6 w-32 bg-gray-200 rounded animate-pulse"></div>
              </CardHeader>
              <CardContent>
                <div className="h-64 bg-gray-200 rounded animate-pulse"></div>
              </CardContent>
            </Card>
          </div>
        </div>
      </div>
    </div>
  );
};


