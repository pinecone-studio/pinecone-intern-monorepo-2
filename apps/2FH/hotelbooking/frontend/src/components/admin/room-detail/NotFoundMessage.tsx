/* eslint-disable  */
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { Search } from 'lucide-react';

export const NotFoundMessage = () => {
  return (
    <div className="min-h-screen bg-gray-50 p-6">
      <div className="max-w-6xl mx-auto">
        <Card>
          <CardHeader>
            <CardTitle className="flex items-center gap-2 text-gray-600">
              <Search className="h-5 w-5" />
              Room Not Found
            </CardTitle>
          </CardHeader>
          <CardContent>
            <p className="text-gray-700">The room you're looking for could not be found.</p>
          </CardContent>
        </Card>
      </div>
    </div>
  );
};
