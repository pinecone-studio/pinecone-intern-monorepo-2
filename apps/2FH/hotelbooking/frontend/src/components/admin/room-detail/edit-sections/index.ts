export { BasicInfoSection } from './BasicInfoSection';
export { AmenitiesSection } from './AmenitiesSection';
export { ImagesSection } from './ImagesSection';
export { DetailsSection } from './DetailsSection';


