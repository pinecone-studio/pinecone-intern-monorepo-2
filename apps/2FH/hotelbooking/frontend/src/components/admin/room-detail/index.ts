export { RoomInfoCard } from './RoomInfoCard';
export { RoomAmenitiesCard } from './RoomAmenitiesCard';
export { RoomImagesCard } from './RoomImagesCard';
export { LoadingSkeleton } from './LoadingSkeleton';
export { ErrorMessage } from './ErrorMessage';
export { NotFoundMessage } from './NotFoundMessage';
export { RoomDetailPage } from './RoomDetailPage';
export { EditRoomModal } from './EditRoomModal';
