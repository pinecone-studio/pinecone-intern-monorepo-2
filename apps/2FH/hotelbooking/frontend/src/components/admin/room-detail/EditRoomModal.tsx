/* eslint-disable  */
'use client';
import { useState } from 'react';
import { useUpdateRoomMutation } from '@/generated';
import { Button } from '@/components/ui/button';
import { Dialog, DialogContent, DialogHeader, DialogTitle, DialogFooter } from '@/components/ui/dialog';
import { BasicInfoSection, AmenitiesSection, ImagesSection, DetailsSection } from './edit-sections';

interface EditRoomModalProps {
  room: any;
  section: 'basic' | 'amenities' | 'images' | 'details';
  isOpen: boolean;
  onOpenChange: (open: boolean) => void;
  refetch: () => Promise<any>;
  roomId: string;
}

export const EditRoomModal = ({ room, section, isOpen, onOpenChange, refetch, roomId }: EditRoomModalProps) => {
  const [updateRoom, { loading: updateLoading }] = useUpdateRoomMutation();

  const normalizeStatus = (status: any) => {
    if (!status) return 'available';
    const statusStr = status.toString().toLowerCase();
    const validStatuses = ['available', 'booked', 'cancelled', 'completed', 'pending'];
    return validStatuses.includes(statusStr) ? statusStr : 'available';
  };

  const [formData, setFormData] = useState(() => ({
    ...room,
    status: normalizeStatus(room.status),
    internet: room.internet || [],
    foodAndDrink: room.foodAndDrink || [],
    bedRoom: room.bedRoom || [],
    bathroom: room.bathroom || [],
    accessibility: room.accessibility || [],
    entertainment: room.entertainment || [],
    other: room.other || [],
    roomInformation: room.roomInformation || [],
    imageURL: room.imageURL || [],
  }));

  const handleInputChange = (field: string, value: any) => {
    setFormData((prev: any) => ({
      ...prev,
      [field]: value,
    }));
  };

  const handleSave = async () => {
    try {
      const input: any = {};

      input.bedNumber = formData.bedNumber || room.bedNumber;
      const currentStatus = formData.status || room.status;
      input.status = normalizeStatus(currentStatus);
      if (section === 'basic') {
        if (formData.name !== undefined) input.name = formData.name;
        if (formData.pricePerNight !== undefined) input.pricePerNight = formData.pricePerNight;
        if (formData.typePerson !== undefined) input.typePerson = formData.typePerson;
        if (formData.roomInformation !== undefined) input.roomInformation = formData.roomInformation;
      } else if (section === 'amenities') {
        if (formData.internet !== undefined) input.internet = formData.internet;
        if (formData.foodAndDrink !== undefined) input.foodAndDrink = formData.foodAndDrink;
        if (formData.bedRoom !== undefined) input.bedRoom = formData.bedRoom;
        if (formData.bathroom !== undefined) input.bathroom = formData.bathroom;
        if (formData.accessibility !== undefined) input.accessibility = formData.accessibility;
        if (formData.entertainment !== undefined) input.entertainment = formData.entertainment;
        if (formData.other !== undefined) input.other = formData.other;
      } else if (section === 'images') {
        if (formData.imageURL !== undefined) input.imageURL = formData.imageURL;
      } else if (section === 'details') {
        if (formData.roomInformation !== undefined) input.roomInformation = formData.roomInformation;
      }
      const result = await updateRoom({
        variables: {
          updateRoomId: roomId,
          input,
        },
      });
      if (result.data?.updateRoom) {
        await refetch();
        onOpenChange(false);
      } else {
        console.error('Update failed: No data returned');
      }
    } catch (error: any) {
      console.error('Error updating room:', error);
      console.error('Error details:', error.message);
      console.error('GraphQL errors:', error.graphQLErrors);
      console.error('Network errors:', error.networkError);
      console.error('Full error object:', error);
    }
  };
  const getSectionComponent = () => {
    let component;
    switch (section) {
      case 'basic':
        component = <BasicInfoSection room={formData} handleInputChange={handleInputChange} />;
        break;
      case 'amenities':
        component = <AmenitiesSection room={formData} handleInputChange={handleInputChange} />;
        break;
      case 'images':
        component = <ImagesSection room={formData} handleInputChange={handleInputChange} />;
        break;
      case 'details':
        component = <DetailsSection room={formData} handleInputChange={handleInputChange} />;
        break;
      default:
        component = null;
    }
    return component || null;
  };
  const getSectionTitle = () => {
    const titleMap = {
      basic: 'Basic Information',
      amenities: 'Amenities',
      images: 'Images',
      details: 'Details',
    };
    return titleMap[section] || 'Edit Room';
  };
  return (
    <Dialog open={isOpen} onOpenChange={onOpenChange}>
      <DialogContent className="max-w-4xl max-h-[90vh] overflow-y-auto" data-testid="edit-room-modal">
        <DialogHeader>
          <DialogTitle data-testid="edit-room-modal-title">Edit {getSectionTitle()}</DialogTitle>
        </DialogHeader>

        <div className="space-y-6 py-4">
          <div className="min-h-[400px]" data-testid="edit-room-modal-content">
            {getSectionComponent()}
          </div>
        </div>

        <DialogFooter className="flex justify-end gap-3 pt-4 border-t">
          <Button variant="outline" onClick={() => onOpenChange(false)} disabled={updateLoading} data-testid="edit-room-modal-cancel">
            Cancel
          </Button>
          <Button onClick={handleSave} className="bg-blue-600 hover:bg-blue-700" disabled={updateLoading} data-testid="edit-room-modal-save">
            {updateLoading ? 'Saving...' : 'Save Changes'}
          </Button>
        </DialogFooter>
      </DialogContent>
    </Dialog>
  );
};
