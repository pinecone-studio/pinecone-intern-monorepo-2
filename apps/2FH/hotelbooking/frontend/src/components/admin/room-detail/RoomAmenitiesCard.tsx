'use client';
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { Button } from '@/components/ui/button';
import { Wifi, Utensils, Bed, Accessibility, Gamepad2, Coffee } from 'lucide-react';
import { EditRoomModal } from './EditRoomModal';

interface RoomAmenitiesCardProps {
  room: any;
  editModalState: {
    isOpen: boolean;
    section: 'basic' | 'amenities' | 'images' | 'details';
  };
  setEditModalState: (_state: { isOpen: boolean; section: 'basic' | 'amenities' | 'images' | 'details' }) => void;
  refetch: () => Promise<any>;
  roomId: string;
}

export const RoomAmenitiesCard = ({ room, editModalState, setEditModalState, refetch, roomId }: RoomAmenitiesCardProps) => {
  // Create a comprehensive list of all amenities with their values and labels
  const getAllAmenities = () => {
    const amenityCategories = [
      { key: 'internet', label: 'Internet', icon: <Wifi className="h-4 w-4" /> },
      { key: 'foodAndDrink', label: 'Food & Drink', icon: <Utensils className="h-4 w-4" /> },
      { key: 'bedRoom', label: 'Bedroom', icon: <Bed className="h-4 w-4" /> },
      { key: 'bathroom', label: 'Bathroom', icon: <Coffee className="h-4 w-4" /> },
      { key: 'accessibility', label: 'Accessibility', icon: <Accessibility className="h-4 w-4" /> },
      { key: 'entertainment', label: 'Entertainment', icon: <Gamepad2 className="h-4 w-4" /> },
      { key: 'other', label: 'Other', icon: <Coffee className="h-4 w-4" /> },
    ];

    const allAmenities: any[] = [];

    amenityCategories.forEach((category) => {
      const categoryValue = room[category.key];
      if (categoryValue) {
        // Handle both single values and arrays
        const values = Array.isArray(categoryValue) ? categoryValue : [categoryValue];
        values.forEach((value: string) => {
          if (value) {
            allAmenities.push({
              key: `${category.key}-${value}`,
              category: category.label,
              value: value,
              icon: category.icon,
            });
          }
        });
      }
    });

    return allAmenities;
  };

  const amenities = getAllAmenities();

  return (
    <Card>
      <CardHeader className="flex flex-row items-center justify-between">
        <CardTitle>Room Amenities</CardTitle>
        <EditRoomModal
          room={room}
          section="amenities"
          isOpen={editModalState.isOpen && editModalState.section === 'amenities'}
          onOpenChange={(open) => setEditModalState({ isOpen: open, section: 'amenities' })}
          refetch={refetch}
          roomId={roomId}
        />
        <Button size="sm" className="bg-blue-600 hover:bg-blue-700 text-white" onClick={() => setEditModalState({ isOpen: true, section: 'amenities' })}>
          Edit
        </Button>
      </CardHeader>
      <CardContent>
        {amenities.length === 0 ? (
          <div className="text-center py-8 text-gray-500">No amenities configured for this room</div>
        ) : (
          <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
            {amenities.map((amenity) => (
              <div key={amenity.key} className="flex items-center gap-3 p-3 border rounded-lg">
                <div className="text-blue-600">{amenity.icon}</div>
                <div className="flex-1">
                  <div className="font-medium text-gray-900">{amenity.value.replace(/_/g, ' ').replace(/\b\w/g, (l: string) => l.toUpperCase())}</div>
                  <div className="text-sm text-gray-600">{amenity.category}</div>
                </div>
              </div>
            ))}
          </div>
        )}
      </CardContent>
    </Card>
  );
};
