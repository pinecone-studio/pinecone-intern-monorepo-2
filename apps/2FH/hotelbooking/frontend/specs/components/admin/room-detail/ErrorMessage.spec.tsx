import React from 'react';
import { render, screen } from '@testing-library/react';
import { ErrorMessage } from '@/components/admin/room-detail/ErrorMessage';

describe('ErrorMessage', () => {
  it('renders error message correctly', () => {
    const errorMessage = 'Something went wrong';
    render(<ErrorMessage message={errorMessage} />);

    expect(screen.getByText('Error')).toBeInTheDocument();
    expect(screen.getByText(errorMessage)).toBeInTheDocument();
  });

  it('renders with correct styling classes', () => {
    render(<ErrorMessage message="Test error" />);

    const errorTitle = screen.getByText('Error');
    expect(errorTitle).toHaveClass('text-red-600');

    const container = errorTitle.closest('.min-h-screen');
    expect(container).toHaveClass('bg-gray-50', 'p-6');
  });

  it('displays AlertCircle icon', () => {
    render(<ErrorMessage message="Test error" />);

    const icon = screen.getByText('Error').querySelector('svg');
    expect(icon).toBeInTheDocument();
  });

  it('renders within proper card structure', () => {
    render(<ErrorMessage message="Test error" />);

    expect(screen.getByText('Error')).toBeInTheDocument();
    expect(screen.getByText('Test error')).toBeInTheDocument();

    // Check that the message is within a card structure
    const messageElement = screen.getByText('Test error');
    expect(messageElement).toHaveClass('text-gray-700');
  });

  it('handles empty error message', () => {
    render(<ErrorMessage message="" />);

    expect(screen.getByText('Error')).toBeInTheDocument();
    // Check that the error message paragraph exists (even if empty)
    const errorMessageElement = document.querySelector('.text-gray-700');
    expect(errorMessageElement).toBeInTheDocument();
  });

  it('handles long error messages', () => {
    const longMessage =
      'This is a very long error message that might wrap to multiple lines and should still be displayed correctly without breaking the layout or causing any issues with the component rendering.';
    render(<ErrorMessage message={longMessage} />);

    expect(screen.getByText(longMessage)).toBeInTheDocument();
  });

  it('has proper accessibility structure', () => {
    render(<ErrorMessage message="Test error" />);

    // Check that the error title is properly structured
    const errorTitle = screen.getByText('Error');
    expect(errorTitle).toBeInTheDocument();

    // Check that the message is accessible
    const message = screen.getByText('Test error');
    expect(message).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<ErrorMessage message="Test error message" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
