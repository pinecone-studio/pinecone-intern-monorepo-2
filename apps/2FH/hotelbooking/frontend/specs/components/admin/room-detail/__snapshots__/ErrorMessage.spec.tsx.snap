// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage matches snapshot 1`] = `
<div
  class="min-h-screen bg-gray-50 p-6"
>
  <div
    class="max-w-6xl mx-auto"
  >
    <div
      class="rounded-lg border bg-card text-card-foreground shadow-sm"
    >
      <div
        class="flex flex-col space-y-1.5 p-6"
      >
        <h3
          class="text-2xl font-semibold leading-none tracking-tight flex items-center gap-2 text-red-600"
        >
          <svg
            class="lucide lucide-circle-alert h-5 w-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="12"
            />
            <line
              x1="12"
              x2="12.01"
              y1="16"
              y2="16"
            />
          </svg>
          Error
        </h3>
      </div>
      <div
        class="p-6 pt-0"
      >
        <p
          class="text-gray-700"
        >
          Test error message
        </p>
      </div>
    </div>
  </div>
</div>
`;
